### PR TITLE
Our own proxy for commenting

### PR DESCRIPTION
**Pull request by JustinPinner**
!PLEASE DON'T MERGE YET!

We've seen comment submissions fail due to a breakdown of CORS when the commenter is behind an HTTP 1.0 proxy. This change sets up a route that lets us do the proxying to the discussion API from within our own domain.

However, this code is untested because I've not yet been able to authenticate myself with the identity service when running from localhost and using CODE as the source of articles / comments etc, and so I'm unable to actually try to add a comment.

Please can someone take a look and let me know what glaring errors are lurking here (a lot of this is the result of guesswork on my part) because I may have to merge to master in order to test it on the code environment. In which case I at least need to know for sure that the feature switch logic will work as expected.

Thanks,
Justin.

### CODE REVIEW

File: discussion/app/controllers/DiscussionProxyController.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import conf.Configuration
+import play.api.libs.json.Json
+import play.api.libs.ws.WS
+import play.api.mvc.Action
+
+object DiscussionProxyController extends DiscussionController {
+  def relay(path: String) = Action.async { implicit request =>
+    import play.api.Play.current
+
+    log.info(s"### RELAY!!")
```
Comment (johnduffell) [inline discussion/app/controllers/DiscussionProxyController.scala]: logging is good, but may be worth making the log line more useful?

**Comment by johnduffell** [comment]
it would be quicker and possibly simpler to do all of this on the router rather than hitting discussion backend, look at the nginx.conf, I can help out if you like

File: discussion/app/controllers/DiscussionProxyController.scala (hunk 1 above)
Comment (gklopper) [inline discussion/app/controllers/DiscussionProxyController.scala]: hmmmm

File: discussion/app/controllers/DiscussionProxyController.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    log.info(s"### proxied comment call to: ${apiRoot + path}")
+
+    WS.url(apiRoot + path)
+      .withHeaders(headers: _*)
+      .withMethod(request.method)
+      .withRequestTimeout(timeout)
+      .post(payload).map{ response =>
+        Ok(response.json)
```
Comment (gklopper) [inline discussion/app/controllers/DiscussionProxyController.scala]: You will want cache headers - probably `no-cache`

**Comment by gklopper** [comment]
Lets hear that Nginx plan

**Comment by robertberry** [comment]
Ship it or close it

**Comment by johnduffell** [comment]
I think someone who has some time needs to have a rethink on this to be honest

**Pull request closed without merging**